packages/visual-group/src/encoder/cartesian-encoder.js

```diff
@@ -75,7 +75,10 @@ export default class CartesianEncoder extends VisualEncoder {
             } else {
                 axesCreators.position = this.axisFrom()[type];
             }
-            geomCellAxes[axis] = generateAxisFromMap(axis, axisFields[i], axesCreators, axis === X ? xAxes : yAxes);
+            geomCellAxes[axis] = generateAxisFromMap(axis, axisFields[i], axesCreators, {
```

Comment:
Why is the axis created like this?
Use sanitizer to create a common data structure which then drives the creation of axis from `generateAxisFromMap`